zone/client_packet.cpp

```diff
@@ -16533,3 +16644,306 @@ void Client::ReloadExpansionProfileSetting()
+}
+
+void Client::Handle_OP_GuildTributeOptInOut(const EQApplicationPacket* app)
+{
+	LogTribute("Received OP_Handle_OP_GuildTributeOptInOut of length [{}]", app->size);
+
+	if (app->size != sizeof(GuildTributeOptInOutReq_Struct)) {
+		LogError("Invalid size on Handle_OP_GuildTributeOptInOut packet");
+		return;
+	}
+
+	GuildTributeOptInOutReq_Struct* in = (GuildTributeOptInOutReq_Struct*)app->pBuffer;
+
+	CharGuildInfo gci;
+	guild_mgr.GetCharInfo(in->player, gci);
+
+	guild_mgr.UpdateDbMemberTributeEnabled(GuildID(), gci.char_id, in->tribute_toggle);
+
+	ServerPacket* sout = new ServerPacket(ServerOP_GuildTributeOptInToggle, sizeof(GuildTributeMemberToggle));
+	GuildTributeMemberToggle* out = (GuildTributeMemberToggle*)sout->pBuffer;
+
+	out->guild_id = GuildID();
+	out->char_id = gci.char_id;
+	strncpy(out->player_name, in->player, strlen(in->player));
+	out->tribute_toggle = in->tribute_toggle;
+	out->command = in->command;
+
+	worldserver.SendPacket(sout);
+
+	safe_delete(sout);
+}
+
+void Client::Handle_OP_GuildTributeSaveActiveTributes(const EQApplicationPacket* app)
+{
+	LogTribute("Received OP_Handle_OP_GuildTributeSaveActive of length [{}]", app->size);
+
+	if (app->size != sizeof(GuildTributeSaveActive_Struct)) {
+		LogError("Invalid size on Handle_OP_GuildTributeOptInOut packet");
+		return;
+	}
+
+	GuildTributeSaveActive_Struct* data = (GuildTributeSaveActive_Struct*)app->pBuffer;
+	auto guild = guild_mgr.GetGuildByGuildID(GuildID());
+	if (guild) {
+		GuildTributesRepository::GuildTributes gt{};
+
+		gt.guild_id = GuildID();
+		gt.tribute_id_1 = data->tribute_id_1;
+		gt.tribute_id_2 = data->tribute_id_2;
+		gt.tribute_id_1_tier = data->tribute_1_tier;
+		gt.tribute_id_2_tier = data->tribute_2_tier;
+		gt.enabled = 0;
+		gt.time_remaining = RuleI(Guild, TributeTime);
+		GuildTributesRepository::ReplaceOne(database, gt);
+
+		guild->tribute.enabled = 0;
+		guild->tribute.id_1 = data->tribute_id_1;
+		guild->tribute.id_2 = data->tribute_id_2;
+		guild->tribute.id_1_tier = data->tribute_1_tier;
+		guild->tribute.id_2_tier = data->tribute_2_tier;
+		guild->tribute.time_remaining = RuleI(Guild, TributeTime);
+
+		ServerPacket* sp = new ServerPacket(ServerOP_GuildTributeUpdate, sizeof(GuildTributeUpdate));
+		GuildTributeUpdate* gtu = (GuildTributeUpdate*)sp->pBuffer;
+
+		gtu->guild_id = GuildID();
+		gtu->tribute_id_1 = data->tribute_id_1;
+		gtu->tribute_id_2 = data->tribute_id_2;
+		gtu->tribute_id_1_tier = data->tribute_1_tier;
+		gtu->tribute_id_2_tier = data->tribute_2_tier;
+		gtu->enabled = 0;
+		gtu->favor = guild->tribute.favor;
+		gtu->time_remaining = RuleI(Guild, TributeTime);
+
+		worldserver.SendPacket(sp);
+		safe_delete(sp);
+	}
+}
+
+void Client::Handle_OP_GuildTributeToggle(const EQApplicationPacket* app)
+{
+	LogTribute("Received OP_Handle_OP_GuildTributeToggle of length [{}]", app->size);
+
+	if (app->size != sizeof(GuildTributeToggleReq_Struct)) {
+		LogError("Invalid size on Handle_OP_GuildModifyBenefits packet");
```

Comment:
Return early here and remove the `else {`

Reply:
Done